libs/agentc_core/agentc_core/catalog/descriptor.py

```diff
@@ -47,10 +48,8 @@ class CatalogDescriptor(pydantic.BaseModel):
 
     kind: CatalogKind = pydantic.Field(description="The type of items within the catalog.")
 
-    embedding_model: str = pydantic.Field(
-        description="The sentence-transformers embedding model used to generate the vector representations "
-        "of each catalog entry.",
-        examples=["sentence-transformers/all-MiniLM-L12-v2"],
+    embedding_model: EmbeddingModel = pydantic.Field(
+        description="Embedding model used for all descriptions in the catalog.",
```

Comment:
```suggestion
        description="Embedding model used to generate embedding for tool/prompt descriptions to store in the catalogs.",
```
something like this would be more suitable